redash/query_runner/mongodb.py

```diff
@@ -180,13 +171,14 @@ def __init__(self, configuration):
     def _get_db(self):
         kwargs = {}
         if self.is_replica_set:
-            db_connection = pymongo.MongoClient(
-                self.configuration["connectionString"],
-                replicaSet=self.configuration["replicaSetName"],
-                readPreference=self.configuration["readPreference"]
-            )
-        else:
-            db_connection = pymongo.MongoClient(self.configuration["connectionString"])
+            kwargs["replicaSet"] = self.configuration["replicaSetName"]
+            readPreference = self.configuration.get("readPreference")
```

Comment:
This was actually a Key error since there was no "readPreference" to execute `self.configuration["readPreference"]`.

**Before**
<img width="1778" alt="Screen Shot 2020-06-05 at 11 24 27" src="https://user-images.githubusercontent.com/3356951/83890912-cd1a4900-a722-11ea-84f2-560a720a7e9f.png">

**After**
<img width="1327" alt="Screen Shot 2020-06-05 at 11 49 04" src="https://user-images.githubusercontent.com/3356951/83890932-d3a8c080-a722-11ea-9e59-2211e0b6c985.png">